gateway/src/node/client_handling/websocket/connection_handler.rs

```diff
@@ -378,12 +379,29 @@ where
                     enc_address,
                     iv,
                 } => {
+                    *final_receiver = Some(mix_receiver);
                     self.handle_authenticate(address, enc_address, iv, mix_sender)
                         .await
                 }
                 ClientControlRequest::RegisterHandshakeInitRequest { data } => {
+                    *final_receiver = Some(mix_receiver);
                     self.handle_register(data, mix_sender).await
                 }
+                ClientControlRequest::BandwidthCredential { enc_credential } => {
+                    let credential = match ClientControlRequest::try_from_enc_bandwidth_credential(
+                        enc_credential,
+                        self.shared_key.as_ref().unwrap(),
+                    ) {
+                        Ok(c) => c,
+                        Err(e) => {
+                            return ServerResponse::Error {
+                                message: e.to_string(),
+                            }
+                        }
+                    };
+                    let status = credential.verify(&self.aggregated_verification_key).await;
```

Comment:
This function doesn't need to be async anymore

Reply:
Correct, I haven't seen that. I'll remove the async